include/nbl/builtin/glsl/ext/RadixSort/default_scatter.comp

```diff
@@ -0,0 +1,142 @@
+{
+	uvec2 in_values[];
+};
+
+#define _NBL_GLSL_EXT_RADIXSORT_SCATTER_INPUT_DESCRIPTOR_DEFINED_
+#endif
+
+#ifndef _NBL_GLSL_EXT_RADIXSORT_SCATTER_OUTPUT_DESCRIPTOR_DEFINED_
+
+layout (set = _NBL_GLSL_EXT_RADIXSORT_SCATTER_OUTPUT_SET_DEFINED_, binding = _NBL_GLSL_EXT_RADIXSORT_SCATTER_OUTPUT_BINDING_DEFINED_, std430) writeonly buffer out_buffer
+{
+	uvec2 out_values[];
+};
+
+#define _NBL_GLSL_EXT_RADIXSORT_SCATTER_OUTPUT_DESCRIPTOR_DEFINED_
+#endif
+
+#ifndef _NBL_GLSL_EXT_RADIXSORT_SCATTER_INPUT_HISTOGRAM_DESCRIPTOR_DEFINED_
+
+layout (set = _NBL_GLSL_EXT_RADIXSORT_SCATTER_INPUT_HISTOGRAM_SET_DEFINED_, binding = _NBL_GLSL_EXT_RADIXSORT_SCATTER_INPUT_HISTOGRAM_BINDING_DEFINED_, std430) readonly buffer histogram_buffer
+{
+	uint histogram[];
+};
+
+#define _NBL_GLSL_EXT_RADIXSORT_SCATTER_INPUT_HISTOGRAM_DESCRIPTOR_DEFINED_
+#endif
+
+#include "nbl/builtin/glsl/ext/RadixSort/parameters_struct.glsl"
+#include "nbl/builtin/glsl/ext/RadixSort/parameters.glsl"
+
+#ifndef _NBL_GLSL_EXT_RADIXSORT_PUSH_CONSTANTS_DEFINED_
+
+layout(push_constant) uniform PushConstants
+{
+	layout (offset = 0) nbl_glsl_ext_RadixSort_Parameters_t params;
+} pc;
+
+#define _NBL_GLSL_EXT_RADIXSORT_PUSH_CONSTANTS_DEFINED_
+#endif
+
+#ifndef _NBL_GLSL_EXT_RADIXSORT_GET_PARAMETERS_DEFINED_
+
+nbl_glsl_ext_RadixSort_Parameters_t nbl_glsl_ext_RadixSort_getParameters()
+{
+	return pc.params;
+}
+
+#define _NBL_GLSL_EXT_RADIXSORT_GET_PARAMETERS_DEFINED_
+#endif
+
+shared uint global_histogram[NUM_BUCKETS];
+
+#ifndef _NBL_GLSL_EXT_RADIXSORT_LOAD_GLOBAL_HISTOGRAM_DEFINED_
+
+void nbl_glsl_ext_RadixSort_loadGlobalHistogram()
+{
+	if (gl_LocalInvocationIndex < NUM_BUCKETS)
+		global_histogram[gl_LocalInvocationIndex] = histogram[gl_LocalInvocationIndex * gl_NumWorkGroups.x + gl_WorkGroupID.x];
+	barrier();
+}
+
+#define _NBL_GLSL_EXT_RADIXSORT_LOAD_GLOBAL_HISTOGRAM_DEFINED_
+#endif
+
+#include <nbl/builtin/glsl/limits/numeric.glsl>
+
+#ifndef _NBL_GLSL_EXT_RADIXSORT_SCATTER_GET_PADDED_DATA_DEFINED_
+
+uvec2 nbl_glsl_ext_RadixSort_getPaddedData(in uint idx)
+{
+	uvec2 data = uvec2(UINT_MAX);
+	if (idx < nbl_glsl_ext_RadixSort_Parameters_t_getElementCountTotal())
+		data = in_values[gl_GlobalInvocationID.x];
+
+	return data;
+}
+
+#define _NBL_GLSL_EXT_RADIXSORT_SCATTER_GET_PADDED_DATA_DEFINED_
+#endif
+
+#ifndef _NBL_GLSL_EXT_RADIXSORT_SCATTER_SET_DATA_DEFINED_
+
+void nbl_glsl_ext_RadixSort_setData(in uint idx, in uvec2 data)
+{
+	if (idx < nbl_glsl_ext_RadixSort_Parameters_t_getElementCountTotal())
+		out_values[idx] = data;
+}
+
+#define _NBL_GLSL_EXT_RADIXSORT_SCATTER_SET_DATA_DEFINED_
+#endif
+
+#include "nbl/builtin/glsl/ext/RadixSort/radix_sort.glsl"
+
+#ifndef _NBL_GLSL_EXT_RADIXSORT_SCATTER_MAIN_DEFINED_
+
+void main()
+{	
+	nbl_glsl_ext_RadixSort_loadGlobalHistogram();
+
+	uvec2 data = nbl_glsl_ext_RadixSort_getPaddedData(gl_GlobalInvocationID.x);
+	uint digit = nbl_glsl_ext_RadixSort_extractDigit(data.x);
```

Comment:
due to https://github.com/Devsh-Graphics-Programming/Nabla/pull/75/files?file-filters%5B%5D=.comp&file-filters%5B%5D=.glsl&file-filters%5B%5D=.txt#r608186550

I'd do
```glsl
const uvec2 data = nbl_glsl_ext_RadixSort_getPaddedData(gl_GlobalInvocationID.x);
const uint digit = nbl_glsl_ext_RadixSort_extractDigit(data.x);

const uint offset = nbl_glsl_ext_RadixSort_loadGlobalHistogram(digit);
const uint scatter_idx = nbl_glsl_ext_RadixSort_workgroupCompact(digit,false);

const uint global_scatter_idx = offset+scatter_idx;
nbl_glsl_ext_RadixSort_setData(global_scatter_idx, data);
```